src/base/msg_buffer.rs

```diff
@@ -0,0 +1,21 @@
+macro_rules! buffer_of_size {
+    ($size: expr) => {
+        [0u8; $size]
+    };
+}
```

Comment:
This macro seems to only make the syntax *more* verbose.

```rust
buffer_of_size(100)
```

vs.

```rust
[0u8; 100]
```

Or are we really worried about the duplication of `0u8`?

Reply:
The user of these two functions has no idea the object they're being returned. To them there's equally a concept of a "Message Buffer" and unless you go and read the code and see that it is in fact backed by a `[0u8; 100]`

So - in the future, when we find a **better way** to implement Message Buffers, we change it in one place and the whole program benefits from it

This is in contrast to a situation in which we write `[0u8; 100]` in a thousand places and then later think "Ah, we could've communicated with the USB bus a lot better and more efficiently" and then need to refactor all thousand of the places

Verbosity wasn't the goal here - It's more of "We're very bad at this, at this stage. This is probably a terrible implementation and we just don't know it yet - let's make it so we can make mistakes and then most easily fix those mistakes"

Reply:
Or - TL;DR; - I spent ~8 hours trying to figure out how to write to the USB bus and I'm still not happy with the implementation. I'm sure I'll change my mind in a week, and when I do so I'd love to refactor it quickly!

Reply:
Let me try to give you my heuristic for macros vs constant functions and see if you like it.

I use macros when I'm actually operating on syntax. When I have the urge to actually **write** something redundant, but want to avoid the redundancy.

I use constant functions (or functions in general) when I want to create an abstraction and hide some implementation details.

Now that Rust has [const generics](https://blog.rust-lang.org/2021/02/26/const-generics-mvp-beta.html) this should be doable!

```rust
const fn buffer_of_size<const N: usize>() -> [u8; N] {
    [0u8; N]
}

let mut msg_buf = buffer_of_size::<100>();
```